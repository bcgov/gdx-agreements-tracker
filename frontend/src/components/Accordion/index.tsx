import * as React from "react";
import { Accordion as MUIAccordion } from "@mui/material/";
import AccordionSummary from "@mui/material/AccordionSummary";
import AccordionDetails from "@mui/material/AccordionDetails";
import Typography from "@mui/material/Typography";
import ExpandMoreIcon from "@mui/icons-material/ExpandMore";
import { styled } from "@mui/material";
import bcgovTheme from "../../bcgovTheme";

export const Accordion = ({
  children,
  sectionTitle,
}: {
  children: JSX.Element | JSX.Element[];
  sectionTitle: string;
}) => {
  const StyledAccordionSummary = styled(AccordionSummary)({
    backgroundColor: "#f3f3f3",
  });

  const StyledTypography = styled(Typography)({
    color: bcgovTheme.palette.primary.main,
    fontWeight: "bold",
  });

  return (
    <MUIAccordion defaultExpanded>
      <StyledAccordionSummary
        expandIcon={<ExpandMoreIcon />}
        aria-controls="panel-content"
        aria-label="button"
      >
        <StyledTypography>{sectionTitle}</StyledTypography>
      </StyledAccordionSummary>
      <AccordionDetails>{children}</AccordionDetails>
    </MUIAccordion>
  );
};
