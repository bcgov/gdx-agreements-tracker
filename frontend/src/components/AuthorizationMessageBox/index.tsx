import { FormHelperText } from "@mui/material";

export const AuthorizationMessageBox = ({ message }: { message: string }) => {
  return <FormHelperText>{message}</FormHelperText>;
};
