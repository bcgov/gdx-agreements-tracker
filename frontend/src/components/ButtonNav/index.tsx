import React from "react";
import { Link, useMatch, useResolvedPath } from "react-router-dom";
import type { LinkProps } from "react-router-dom";

/**
 * Returns react router link with active class if matches route.
 *
 * @param {LinkProps} root0 Link properties.
 * @param {ReactNode | undefined} root0.children Child nodes.
 * @param {To} root0.to Destination.
 * @returns {React.ReactNode}
 */
export const ButtonNav = ({ children, to, ...props }: LinkProps) => {
  const resolved = useResolvedPath(to);
  const match = useMatch({ path: resolved.pathname, end: true });
  if (match) {
    props.className = `${props.className ?? ""} active`;
  }
  return (
    <Link to={to} {...props}>
      {children}
    </Link>
  );
};

export default ButtonNav;
