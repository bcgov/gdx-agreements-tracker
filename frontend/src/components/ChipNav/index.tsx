import * as React from "react";
import { styled } from "@mui/material/styles";
import Chip from "@mui/material/Chip";
import { IChipNav } from "../../types";
import { Box, Paper } from "@mui/material";
import { NavLink } from "react-router-dom";
import bcgovTheme from "../../bcgovTheme";

export const ChipNav = ({
  navLinks,
  navLinksRight = [],
}: {
  navLinks: IChipNav[];
  navLinksRight?: IChipNav[];
}) => {
  const StyledListItem = styled("li")(({ theme }) => ({
    margin: theme.spacing(0.5),
  }));

  const StyledNavLink = styled(NavLink)(() => ({
    textDecoration: "none",
  }));

  /**
   * Renders a single ChipNav link.
   *
   * @param   {IChipNav}    link The ChipNav link to render.
   * @returns {JSX.Element}
   */
  const renderNavLink = (link: IChipNav): JSX.Element => {
    return (
      <StyledListItem key={link.key}>
        <StyledNavLink to={link.url} end>
          {({ isActive }) => (
            <Chip
              sx={{
                backgroundColor: isActive ? "#D3D3D3" : bcgovTheme.palette.primary.main,
                color: isActive ? "#000" : "#FFF",
              }}
              label={link.name}
            />
          )}
        </StyledNavLink>
      </StyledListItem>
    );
  };

  return (
    <Paper
      elevation={3}
      sx={{
        display: "flex",
        justifyContent: "space-between",
        flexWrap: "wrap",
        p: 0.5,
        mb: 2,
        mt: 0,
      }}
    >
      <Box sx={{ display: "flex", listStyle: "none" }}>{navLinks.map(renderNavLink)}</Box>
      {navLinksRight.length > 0 && (
        <Box sx={{ display: "flex", listStyle: "none" }}>{navLinksRight.map(renderNavLink)}</Box>
      )}
    </Paper>
  );
};
