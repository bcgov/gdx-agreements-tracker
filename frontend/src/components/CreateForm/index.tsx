import React from "react";
import { Box, Button } from "@mui/material";
import { Form, Formik, FormikHelpers, FormikValues } from "formik";
import { FormLayout } from "../GDXForm/FormLayout";
import { FormInput } from "../FormInput";
import { IEditFields } from "../../types";

/**
 * Destructured as:
 * @param    {​​​​​​Object[]}​​​​​​ initialValues - Contains all the initial values for Formik to use
 * @param    {​​​​​​​​​​​​​​​​​​​​string}​​​​​​​​​​​​​​​​​​​​ onSubmit - Handles the submit functionality for the form
 * @param    {Object[]}​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​ editFields - The field properties used to render the fields for an edit form
 * @returns  {React.JSX}
 */

export const CreateForm = ({
  initialValues,
  onSubmit,
  editFields,
}: {
  initialValues: FormikValues;
  // todo: Define a good type. "Any" type temporarily permitted.
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  onSubmit: ((values: unknown, formikHelpers: FormikHelpers<any>) => void | Promise<any>) &
    Function;
  editFields: IEditFields[];
}) => {
  return (
    <Formik initialValues={initialValues} onSubmit={onSubmit}>
      {({ setFieldValue, values, handleChange, dirty }) => {
        return (
          <Form>
            <FormLayout>
              {editFields.map(({ fieldName, fieldType, fieldLabel, width, tableName }) => {
                return (
                  <FormInput
                    setFieldValue={setFieldValue}
                    fieldValue={values?.[fieldName]}
                    fieldName={fieldName}
                    fieldType={fieldType}
                    fieldLabel={fieldLabel}
                    handleChange={handleChange}
                    width={width}
                    key={fieldName}
                    tableName={tableName}
                  />
                );
              })}
            </FormLayout>
            <Box
              m={1}
              display="flex"
              justifyContent="flex-end"
              alignItems="flex-end"
              role={"submit_button"}
            >
              <Button
                type="submit"
                variant="contained"
                color="success"
                disabled={dirty ? false : true}
              >
                Submit
              </Button>
            </Box>
          </Form>
        );
      }}
    </Formik>
  );
};
