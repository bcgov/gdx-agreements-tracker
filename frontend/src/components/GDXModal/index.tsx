import React from "react";
import { Box, Modal, styled } from "@mui/material";
import { FormHeader } from "../GDXForm/FormLayout/FormHeader";

export const GDXModal = ({
  children,
  open,
  handleClose,
  modalTitle,
  handleEditMode,
  editMode,
}: {
  children: JSX.Element;
  open: boolean;
  handleClose: () => void;
  modalTitle: string;
  handleEditMode: Function;
  editMode: boolean;
}) => {
  const StyledModalBox = styled(Box)({
    position: "absolute",
    top: "50%",
    left: "50%",
    transform: "translate(-50%, -50%)",
    backgroundColor: "#FFF",
    width: "80%",
    overflowY: "auto",
    height: "auto",
    maxHeight: "100%",
  });

  const StyledContentBox = styled(Box)({
    padding: "20px",
  });

  const StyledModal = styled(Modal)({
    overflow: "overlay",
  });

  return (
    <>
      <StyledModal
        open={open}
        aria-labelledby="modal-modal-title"
        aria-describedby="modal-modal-description"
        onClose={handleClose}
      >
        <StyledModalBox>
          <FormHeader formTitle={modalTitle} handleEditMode={handleEditMode} editMode={editMode} />
          <StyledContentBox>{children}</StyledContentBox>
        </StyledModalBox>
      </StyledModal>
    </>
  );
};
