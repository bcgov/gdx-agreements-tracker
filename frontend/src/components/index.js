export { ButtonNav } from "./ButtonNav";
export { Debug } from "./Debug";
export { Loader } from "./Loader";
export { Main } from "./Layout/Main";
export { PageLink } from "./PageLink";
export { RowEditDeleteChip } from "./RowEditDeleteChip";
export { Sidebar } from "./Sidebar";
export { DetailsList } from "./DetailsList";
export { Table } from "./Table";
