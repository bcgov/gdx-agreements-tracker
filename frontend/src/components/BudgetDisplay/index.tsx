import { Box, Grid } from "@mui/material";
import { GDXList } from "components/GDXList";
import { Loader } from "components/Loader";
import { useAxios } from "hooks/useAxios";
import React from "react";
import { useQuery } from "react-query";

export const BudgetDisplay = ({ apiUrl }: { apiUrl: string }) => {
  const { axiosAll } = useAxios();

  const getBudgetData = async () => {
    const budget = await axiosAll()
      .get(apiUrl)
      .then((budgetData: any) => {
        return budgetData;
      });
    return budget.data.data;
  };

  // Queries
  const { data, isLoading } = useQuery("contract budget", getBudgetData, {
    // todo: When there is an edit and view form built, reassess these options.
    refetchOnWindowFocus: false,
    retryOnMount: false,
    refetchOnReconnect: false,
    retry: false,
    staleTime: Infinity,
  });

  return (
    <Grid container spacing={2}>
      {!data ? (
        <Loader />
      ) : (
        data.map((budget: any) => {
          return <GDXList data={budget} title={`Fiscal ${budget.fiscal_year}`} />;
        })
      )}
    </Grid>
  );
};