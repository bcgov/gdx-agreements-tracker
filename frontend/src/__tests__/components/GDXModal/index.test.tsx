import React from "react";
import { mount } from "enzyme";
import { GDXModal } from "components/GDXModal";

const editMode = false;
const allowEdit = false;
const allowDelete = false;

const handleEditMode = () => {
  jest.fn();
};
const handleClose = () => {
  jest.fn();
};
const handleDelete = () => {
  jest.fn();
};
const handleFormType = () => {
  jest.fn();
};

describe("Tests different renders of the <Renderer /> component", () => {
  //This function allows you to render the component with different isLoading props to test different outcomes
  const wrapper = ({ open }: { open: boolean }) => {
    return mount(
      <GDXModal
        open={open}
        handleClose={handleClose}
        modalTitle={"test title"}
        handleEditMode={handleEditMode}
        editMode={editMode}
        allowEdit={allowEdit}
        allowDelete={allowDelete}
        handleDelete={handleDelete}
        handleFormType={handleFormType}
      >
        <div>Mock Div</div>
      </GDXModal>
    );
  };

  it("snapshot test for modal open", () => {
    expect(wrapper({ open: true }).html()).toMatchSnapshot();
  });

  it("snapshot test for modal closed", () => {
    expect(wrapper({ open: false }).html()).toMatchSnapshot();
  });
});
