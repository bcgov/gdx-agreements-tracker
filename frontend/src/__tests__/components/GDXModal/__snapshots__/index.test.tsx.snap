// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests different renders of the <Renderer /> component snapshot test for modal closed 1`] = `""`;

exports[`Tests different renders of the <Renderer /> component snapshot test for modal open 1`] = `"<div role=\\"presentation\\" aria-labelledby=\\"modal-modal-title\\" aria-describedby=\\"modal-modal-description\\" class=\\"MuiModal-root css-79ws1d-MuiModal-root\\"><div aria-hidden=\\"true\\" class=\\"MuiBackdrop-root css-i9fmh8-MuiBackdrop-root-MuiModal-backdrop\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\"></div><div tabindex=\\"0\\" data-testid=\\"sentinelStart\\"></div><div class=\\"MuiBox-root css-1ds7qt2\\" tabindex=\\"-1\\"><div><header class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionSticky css-15ta3ox-MuiPaper-root-MuiAppBar-root\\" role=\\"form-header\\"><div class=\\"MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-ftu08t-MuiToolbar-root\\" role=\\"form-header-toolbar\\"><div class=\\"MuiBox-root css-jztgcr\\"><div class=\\"MuiBox-root css-k008qs\\"><h6 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1k1pm2e-MuiTypography-root\\">test title</h6></div><div class=\\"MuiBox-root css-k008qs\\"><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary MuiIconButton-sizeLarge css-1f88mc5-MuiButtonBase-root-MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"HighlightOffIcon\\"><path d=\\"M14.59 8 12 10.59 9.41 8 8 9.41 10.59 12 8 14.59 9.41 16 12 13.41 14.59 16 16 14.59 13.41 12 16 9.41 14.59 8zM12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z\\"></path></svg><span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button></div></div></div></header></div><div class=\\"MuiBox-root css-1rn2qcq\\"><div>Mock Div</div></div></div><div tabindex=\\"0\\" data-testid=\\"sentinelEnd\\"></div></div>"`;
